freecad/gridfinity_workbench/features.py

```diff
@@ -311,7 +313,21 @@ def generate_gridfinity_shape(self, obj: fc.DocumentObject) -> Part.Shape:
         compartment_solid = face.extrude(
             fc.Vector(0, 0, obj.TotalHeight - obj.BaseProfileHeight - obj.BaseWallThickness),
         )
-        fuse_total = fuse_total.cut(feat.make_eco_compartments(obj, layout, compartment_solid))
+
+        # First cut eco compartments to create the interior spaces
+        eco_compartments = feat.make_eco_compartments(obj, layout, compartment_solid)
+        fuse_total = fuse_total.cut(eco_compartments)
+
+        # Now add scoop, but only where eco compartments exist (reversed logic)
+        if obj.Scoop:
+            # Store original UsableHeight and adjust for eco bin's deeper compartment
+            original_usable_height = obj.UsableHeight
+            obj.UsableHeight = obj.TotalHeight - obj.BaseWallThickness
+            scoop = feat.make_scoop(obj)
+            obj.UsableHeight = original_usable_height  # Restore original value

```

Comment:
```suggestion
            scoop = feat.make_scoop(obj, usable_height=obj.TotalHeight - obj.BaseWallThickness)
```
Can you modify the `make_scoop` to take the `UsableHeight` as an argument instead?

Reply:
You can do
```py
def make_scoop(obj, *, usable_height=None):
    if usable_height is None:
        usable_height = obj.UsableHeight
```
If you don't want to add this argument in all the places this function is called

Reply:
Hi, thanks, this makes more sense than my little dance above.

It still works great

CustomEcoBin height 3 with full radius (21mm)

<img width="681" height="536" alt="image" src="https://github.com/user-attachments/assets/253456fb-fd2a-4f6e-ac13-025d9280c4f7" />

With constrained radius of 12 mm

<img width="696" height="564" alt="image" src="https://github.com/user-attachments/assets/b6668ab7-1464-49b5-84a5-815c04fa5515" />